examples: add tenant2-simulator (simulator + tenant Gateway/HTTPRoute…

SB159 opened the pull request:
## Summary
Adds `deployment/examples/tenant2-simulator` (llm-tenant2 example):
- KServe simulator, tenant Gateway/HTTPRoute, OpenShift Route
- API-key secrets (with `kuadrant.io/auth-secret: "true"`)
- README with direct Bash/PowerShell steps

Tested per README: 200 OK on simulator; 429s observed on free tier after policies.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Added example manifests to expose a tenant-scoped simulator via Gateway API (HTTPRoute) and OpenShift Route.
  - Templated host, gateway, namespace, and backend service values to simplify per-tenant configuration.

- Documentation
  - New README detailing end-to-end multi-tenant onboarding: setup, API-key authentication, per-tenant rate limiting, verification, testing (including burst tests), and cleanup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jland-redhat commented on deployment/examples/tenant2-simulator/README.md:
```diff
@@ -0,0 +1,118 @@
+# Tenant example: `llm-tenant2` — simulator model + tenant Gateway + API-key auth + rate limits
+
+This example shows how to stand up a **second tenant** (`llm-tenant2`) on a cluster that already
```
Looks good two request.

Instead of saying "second tenant" I would just use this README to just more genarically say "This is how you would hookup an existing Model/Serving runtime to the centralized MaaS platform"

Especially since this will be the "normal flow" of how it is used. And infact I would add a admonition to let folks know the plan is to eventually automate. Something like
> [NOTE]
> This will be automated in future releases based annotations.

jland-redhat commented on deployment/examples/tenant2-simulator/2-simulator-isvc.yaml:
```diff
@@ -0,0 +1,22 @@
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
```
Lets make the InferenceService a prerequisite for this piece.

The idea with this example is to show them how to hook up an existing ServiceRuntime rather than creating one for them.

jland-redhat commented on deployment/examples/tenant2-simulator/README.md:
```diff
@@ -0,0 +1,118 @@
+
+## What this example creates (tenant-owned)
+- Namespace `llm-tenant2`
+- A tiny **simulator** model via KServe (custom container)
```
Mentioned before but lets make the Serving Runtime a pre-req

SB159 replied:
Fixed

jland-redhat commented on deployment/examples/tenant2-simulator/README.md:
```diff
@@ -0,0 +1,118 @@
+- A tiny **simulator** model via KServe (custom container)
+- A **Gateway + HTTPRoute** scoped to the tenant
+- A public **OpenShift Route** pointing to the tenant gateway service
+- **API-key** Secrets for `free` and `premium` users
```
This will need to change once we implement Bartosz's update, so just leaving this comment to @ him for visibility
@bartoszmajsak

jland-redhat commented on deployment/examples/tenant2-simulator/6-sample-keys.yaml:
```diff
@@ -0,0 +1,55 @@
+    secret.kuadrant.io/user-id: "u-free-1"
+type: Opaque
+stringData:
+  api_key: "freeuser1_key"
```
Do we need different keys? Won't the existing ones work once we connect the route to the policy?

SB159 replied:
Fixed

jland-redhat commented on deployment/examples/tenant2-simulator/README.md:
```diff
@@ -0,0 +1,118 @@
+TENANT_NS=${TENANT_NS:=llm-tenant2}
+GW_NAME=inference-gateway
+
+kustomize build deployment/overlays/openshift | yq '
```
Sorry a little confused, what is this doing exactly?

jland-redhat commented on deployment/examples/tenant2-simulator/3-gateway.yaml:
```diff
@@ -0,0 +1,13 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
```
Question for @bartoszmajsak just to make sure I am correct on this assumption.

We are probably going to want a single gateway in front of all our models not one gateway per model correct?

bartoszmajsak replied:
I would keep it simple and have a single gateway to which we attach models through e.g. `HTTPRoute`s.  Are we missing something by doing it this way?

jland-redhat reviewed (changes requested):
Couple feedback request and a couple change request

SB159 commented:
@jland-redhat I have pushed updates addressing your feedback:

Assume existing ServingRuntime; removed 1-sim-app-configmap.yaml, 2-simulator-isvc.yaml, 3-gateway.yaml.

Removed 6-sample-keys.yaml; README now uses existing keys in $GATEWAY_NS (added tip to list them).

Added parentRefs {name, namespace}, backend via ${BACKEND_SVC}.

5-route.yaml now targets the shared gateway svc inference-gateway-istio in $GATEWAY_NS.

README refocused on “existing runtime + shared Gateway”; updated steps.
